storybookjs/storybook
# Core: Unique outputDir/cacheDir for each configDir

Closes #24082

I think this will also fix this issue: #21510

## What I did

I ensured that storybook when using a cache directory as an output directory, will use a unique sub-dir per storybook configDir.

I also cleaned up a bit of terminal output, there were absolute paths in there, which feels noisy.
It also printed out a message about copying over things from node_modules, which I doubt is useful information to the user, as it was storybook's internal pre-bundled stuff that was getting copied)

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

- Making a static build of any storybook, and check if it work.
- Provide a manual outputDir via a CLI flag and see if it works

- Start a dev sandbox
- check the `node_modules/.cache` directory. It's contents should include:
  ![Screenshot 2023-12-19 at 12 06 09](https://github.com/storybookjs/storybook/assets/3070389/975a0543-23fd-49e1-a8d2-26eca28c11f0)

The long hash is unique per storybook configDir. It does not leak anything personally identifiable.
Here's the code creating it:
https://github.com/storybookjs/storybook/blob/3b94da1a9f4e5942dae067f277c2a4672271cf2e/code/lib/core-server/src/build-dev.ts#L60

- check if the dev version works as expected.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
